clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -355,6 +355,10 @@ public void close() {
     }
 
     public void commitOffsetsAsync(final Map<TopicPartition, OffsetAndMetadata> offsets, OffsetCommitCallback callback) {
+        commitOffsetsAsync(offsets, callback, true);
+    }
+
+    private void commitOffsetsAsync(final Map<TopicPartition, OffsetAndMetadata> offsets, OffsetCommitCallback callback, boolean executeDelayedTasks) {
```

Comment:
One thing I realized while working on this is that we actually shouldn't need to send heartbeats anymore in commitAsync() because commits are themselves treated as heartbeats by the coordinator. That would allow us to remove the `executeDelayedTasks` parameter entirely.